subprojects/messaging/src/main/java/org/gradle/internal/serialize/StackTraceElementPlaceholder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.serialize;
+
+import org.gradle.api.JavaVersion;
+
+import java.io.Serializable;
+
+public class StackTraceElementPlaceholder implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private final String classLoaderName;
+    private final String moduleName;
+    private final String moduleVersion;
+    private final String declaringClass;
+    private final String methodName;
+    private final String fileName;
+    private final int lineNumber;
+
+    public StackTraceElementPlaceholder(StackTraceElement ste) {
```

Comment:
normally the internal `format` field [stored](https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/java/lang/StackTraceElement.java#L426-L427) here whether the module name (and version) should be prepended.
Should we maybe serialize that as well?

E.g. with default `format`:
> at java.base@11.0.11/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

with restored `format` field:
> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

Reply:
I'll see if it can be done without triggering illegal access failures.

Reply:
Unfortunately, that won't fly because this field is private and trying to make it accessible will result in illegal access exceptions. Our tests don't catch that because they deliberately open up java.base to the unnamed module. But we can't expect all clients of the Tooling API to do that.